rns_shell/compositor/Compositor.cpp

```diff
@@ -186,6 +240,16 @@ void Compositor::renderLayerTree() {
 #endif
         RNS_PROFILE_API_OFF("SwapBuffers", windowContext_->swapBuffers(surfaceDamage_));
         client_.didRenderFrame();
+
+#if USE(RNS_SHELL_PARTIAL_UPDATES) && ENABLE(RNS_SHELL_BUFFER_AGE)
+        // Add current frame damage to history.
+        if (frameDamageHistory_.size() >= RNS_SHELL_MAX_FRAME_DAMAGE_HISTORY) {
+          FrameDamages damages = frameDamageHistory_.front();
```

Comment:
Is pop front alone is not enough?

Reply:
Pop front is a void function, and if you want to delete a item from list this good practice is to
1. Get the item using front/back
2. free the item resources ( here we are clearing the vector)
3. then remove item from list.

I didn't get the second point.  We have to do single pop for two reasons.
1. Per frame we need only one slot to insert so one pop is enough when it is full
2. We have to keep max history as much as possible because depending on next frame's buffer age we will need more frames history.